### PR TITLE
docs(dev-server): document hmr utils

### PR DESCRIPTION
**Pull request by alicewriteswrongs**
Add some documentation to `src/dev-server/client/hmr-util.ts` and friends

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Testing

This should be a comment-only change

### CODE REVIEW

File: src/dev-server/client/hmr-util.ts [hunk 1]
```diff
@@ -54,20 +56,54 @@ export const updateCssUrlValue = (versionId: string, fileName: string, oldCss: s
...
+/**
+ * Determine whether a given element is a `<link>` tag pointing to a stylesheet
+ *
+ * @param elm the element to check
+ * @returns whether or not the element is a link stylesheet
+ */
+export const isLinkStylesheet = (elm: Element): boolean =>
   elm.nodeName.toLowerCase() === 'link' &&
   (elm as HTMLLinkElement).href &&
   (elm as HTMLLinkElement).rel &&
```
Comment (rwaskiewicz) [inline src/dev-server/client/hmr-util.ts:68]: Couldn't these be `undefined`? Wouldn't that make it such that the return type of this function could be `boolean | undefined` as a result?

**Review by rwaskiewicz — COMMENTED** [review]
One quick question - I think this might have contributed to the SNC increase, but I'm not entirely sure 🤔

File: src/dev-server/client/hmr-util.ts (hunk 1 above)
Comment (tanner-reits) [reply src/dev-server/client/hmr-util.ts:68]: Could add `!!` to both those checks (which would also evaluate `false` if `href` is an empty string)

Comment (alicewriteswrongs) [reply src/dev-server/client/hmr-util.ts:68]: mm true, I'll throw a `!!` in there

**Review by rwaskiewicz — APPROVED** [review]

**Pull request merged**